lib/inertia_rails/renderer.rb

```diff
@@ -43,15 +43,25 @@ def layout
     end
 
     def computed_props
-      _props = ::InertiaRails.shared_data(@controller).send(prop_merge_method, @props).select do |key, prop|
+      # Cast props to symbol keyed hash before merging so that we have a consistent data structure and
+      # avoid duplicate keys after merging.
+      #
+      # Functionally, this permits using either string or symbol keys in the controller. Since the results
+      # is cast to json, we should treat string/symbol keys as identical.
+      _props = ::InertiaRails.shared_data(@controller).deep_symbolize_keys.send(prop_merge_method, @props.deep_symbolize_keys).select do |key, prop|
         if rendering_partial_component?
           key.to_sym.in? partial_keys
         else
           !prop.is_a?(InertiaRails::Lazy)
         end
       end
 
-      deep_transform_values(_props, lambda {|prop| prop.respond_to?(:call) ? @controller.instance_exec(&prop) : prop }).with_indifferent_access
+      deep_transform_values(
```

Comment:
This is just a formatting change. I just like reading this on multiple lines better.